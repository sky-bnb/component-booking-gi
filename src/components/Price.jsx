import React from 'react';
import styled from 'styled-components';

export const StyledPrice = styled.div`
  display: flex;
  line-height: 10px;
  padding: 0px 16px 0px 0px;
  align-items: baseline;
  align-content: flex-end;
  flex-wrap: wrap;
  justify-content: flex-start;
`;

const StyledPriceAmt = styled.div`
  font-size: 22px;
  overflow-wrap: break-word;
  font-family: 'Quicksand', sans-serif;
  font-weight: 800;
  line-height: 1.44444em;
  color: rgb(72, 72, 72);
  margin: 0px;
`;

const StyledPricePerNight = styled.div`
  font-family: 'Quicksand', sans-serif;
  font-size: 12px;
  font-weight: 600;
  line-height: 1.33333em;
  color: rgb(72, 72, 72);
  margin: 0px;
  line-height: 10px;
  padding-left: 4px;
`;

const Price = (props) => {
  const { price } = props;

  return (
    <StyledPrice>
      <StyledPriceAmt>${price}</StyledPriceAmt>
      <StyledPricePerNight>per night</StyledPricePerNight>
    </StyledPrice>
  );
};

export default Price;
