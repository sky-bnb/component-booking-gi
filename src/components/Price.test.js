import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Price, { StyledPrice } from './Price.jsx';

describe('Price Component', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Price />));

  it('renders the Price component', () => {
    expect(wrapper.find(StyledPrice).length).toBe(1);
  });

  it('matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
