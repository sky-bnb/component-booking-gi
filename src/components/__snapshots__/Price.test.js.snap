// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Component matches the snapshot 1`] = `
<Price__StyledPrice>
  <Price__StyledPriceAmt>
    $
  </Price__StyledPriceAmt>
  <Price__StyledPricePerNight>
    per night
  </Price__StyledPricePerNight>
</Price__StyledPrice>
`;
